testgrid/cmd/entomologist/main_test.go

```diff
@@ -0,0 +1,219 @@
+		},
+		{
+			name: "non-gcs output, reject",
+			args: []string{"--github-org=testorg", "--github-repo=testrepo", "--output=/usr/src/foo/bar"},
+		},
+		{
+			name: "required options",
+			args: []string{
+				"--github-org=testorg",
+				"--github-repo=testrepo",
+				"--output=gs://foo/bar",
+				"--gcs-cred=/usr/foo/creds.json",
+			},
+			expected: &options{
+				organization:    "testorg",
+				repository:      "testrepo",
+				gcsPath:         "gs://foo/bar",
+				gcsCredentials:  "/usr/foo/creds.json",
+				ghProxyEndpoint: "https://api.github.com",
+				ghCredentials:   emptyGitHubCredentials,
+			},
+		},
+		{
+			name: "oneshot with many options",
+			args: []string{
+				"--github-org=testorg",
+				"--github-repo=testrepo",
+				"--github-proxy=https://127.0.0.1:8888",
+				"--github-token=0123456789abcdef",
+				"--output=gs://foo/bar",
+				"--gcs-cred=/usr/foo/creds.json",
+				"--oneshot",
+			},
+			expected: &options{
+				organization:    "testorg",
+				repository:      "testrepo",
+				ghProxyEndpoint: "https://127.0.0.1:8888",
+				ghCredentials:   []byte("0123456789abcdef"),
+				gcsPath:         "gs://foo/bar",
+				gcsCredentials:  "/usr/foo/creds.json",
+				oneshot:         true,
+			},
+		},
+	}
+
+	for _, test := range testCases {
+		flags := flag.NewFlagSet(test.name, flag.ContinueOnError)
+		var actual options
+		err := actual.parseArgs(flags, test.args)
+		switch {
+		case err == nil && test.expected == nil:
+			t.Errorf("%s: failed to return an error", test.name)
+		case err != nil && test.expected != nil:
+			t.Errorf("%s: unexpected error: %v", test.name, err)
+		case test.expected != nil && !reflect.DeepEqual(*test.expected, actual):
+			t.Errorf("%s: actual %#v != expected %#v", test.name, actual, *test.expected)
+		}
+	}
+}
+
+func initFakeGitHub() githubClient {
+	var f fakegithub.FakeClient
+	f.Issues = []github.Issue{
+		{
+			Title:  "Body With No Target; ignore",
+			Number: 2,
+			Body:   "Don't capture this target: it's wrong",
+		},
+		{
+			Title:  "Body with Single Target",
+			Number: 3,
+			Body:   "target:[Organization] Testing Name [sig-testing]",
+		},
+		{
+			Title:  "Body with Multiple Targets",
+			Number: 4,
+			Body:   "target://project:test\r\ntarget://project:example",
+		},
+		{
+			Title:  "Tolerate Whitespace",
+			Number: 5,
+			Body:   "target://project:first\r\ntarget://project:second\ntarget: //project:third",
+		},
+		{
+			Title:  "Target in Comment",
+			Number: 6,
+		},
+		{
+			Title:  "Multiple Comments",
+			Number: 7,
+		},
+		{
+			Title:       "Pull Request; ignore",
+			Number:      8,
+			PullRequest: &struct{}{},
+			Body:        "target://project:no",
+		},
+	}
+
+	f.IssueComments = map[int][]github.IssueComment{
+		6: {
+			{Body: "target://project:test"},
+		},
+		7: {
+			{Body: "target://project:first\r\ntarget://project:second"},
+			{Body: "target://project:third"},
+			{Body: "This is a false target: it's not at the beginning of the line"},
```

Comment:
can you add test cases for
- `target:   leading space is trimmed`
- `target:trailing space is trimmed    ` (which apparently the markdown renderer is trimming _right now_

Reply:
Added in the test's comments for clarity